### ISSUE TITLE
`packageExtensionsChecksum` depends on ordering of package extensions

### ISSUE THREAD

**Issue by chris-olszewski** [pnpm/pnpm#6824]
If one changes the ordering of packages in `packageExtensions` this changes the checksum encoded in the lockfile. 
```
$ git diff
diff --git a/package.json b/package.json
index baffd91..27fa805 100644
--- a/package.json
+++ b/package.json
@@ -16,12 +16,12 @@
   },
   "pnpm": {
     "packageExtensions": {
-      "is-even": {
+      "is-odd": {
         "peerDependencies": {
             "is-number": "*"
         }
       },
-      "is-odd": {
+      "is-even": {
         "peerDependencies": {
             "is-number": "*"
         }
diff --git a/pnpm-lock.yaml b/pnpm-lock.yaml
index 041f5d8..a16d1d8 100644
--- a/pnpm-lock.yaml
+++ b/pnpm-lock.yaml
@@ -4,7 +4,7 @@ settings:
   autoInstallPeers: true
   excludeLinksFromLockfile: false
 
-packageExtensionsChecksum: 5232a40b34ff84131b5d75899b9b5643
+packageExtensionsChecksum: a6d2c3796be1276b05998144be390010
 
 dependencies:
   is-even:
```

This can cause issues for pragmatic altering of `package.json` as it imposes that the order of a map be kept.

### pnpm version:
`8.6.8`

### Code to reproduce the issue:
All that's needed is a [`package.json`](https://gist.github.com/chris-olszewski/05bae929409eaf9a586a4e9d6bbc978c) with multiple package extensions.
 - `pnpm install` to generate lockfile
 - edit `package.json` to change the order of the patched dependencies
 - `pnpm install` and observe that `packageExtensionsChecksum` changes

### Expected behavior:

`packageExtensionChecksum` should remain stable regardless of field ordering

### Actual behavior:

`packageExtensionChecksum` changes with field reordering

### Additional information:

 - `node -v` prints: `v16.18.1`
 - Windows, macOS, or Linux?: macOS

**Comment by laverdet** [comment]
Looks like this landed in v8.6.10. Probably should have been major semver since it leaves around outstanding changes to the lock file.

**Comment by gerbyzation** [comment]
Yup, I had this blow up some pipelines: https://github.com/pnpm/pnpm/pull/6839#issuecomment-1655469125
